cmd/print.go

```diff
@@ -43,55 +46,139 @@ var (
 	// allNamespaces indicates whether all namespaces should be used. Value assigned via
 	// --all-namespaces/-A flag.
 	allNamespaces bool
-)
 
-// printCmd represents the print command. It prints HTTPRoutes and Gateways
-// generated from Ingress resources.
-var printCmd = &cobra.Command{
-	Use:   "print",
-	Short: "Prints HTTPRoutes and Gateways generated from Ingress resources",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		resourcePrinter, err := getResourcePrinter(outputFormat)
+	// resourcePrinter determines how resource objects are printed out
+	resourcePrinter printers.ResourcePrinter
+
+	// Only resources that matches this filter will be processed.
+	namespaceFilter string
+}
+
+// PrintGatewaysAndHttpRoutes performs necessary steps to digest and print
+// converted Gateways and HTTP Routes. The steps includes reading from the source,
+// construct ingresses, convert them, then print them out.
+func (pr *PrintRunner) PrintGatewaysAndHttpRoutes(cmd *cobra.Command, args []string) error {
+	err := pr.initializeResourcePrinter()
+	if err != nil {
+		fmt.Printf("Failed to initialize resrouce printer: %v", err)
+		os.Exit(1)
```

Comment:
since we move this to a RunE cobra function we should just return errors
`fmt.errorf` maybe?

ditto for all os.exit in this file